src/pluto/components/corrections_report.py

```diff
@@ -0,0 +1,143 @@
+import streamlit as st
+import pandas as pd
+import numpy as np
+import plotly.express as px
+from st_aggrid import AgGrid
+from src.constants import COLOR_SCHEME
+
+
+class CorrectionsReport:
+    def __init__(self, data) -> None:
+        self.applied_corrections = data["pluto_corrections_applied"]
+        self.not_applied_corrections = data["pluto_corrections_not_applied"]
+        self.version_dropdown = np.insert(
+            np.flip(np.sort(data["pluto_corrections"].version.dropna().unique())),
+            0,
+            "All",
+        )
+
+    def __call__(self):
+        st.header("Manual Corrections")
+
+        st.markdown(
+            """
+            PLUTO is created using the best available data from a number of city agencies. To further
+            improve data quality, the Department of City Planning (DCP) applies changes to selected field
+            values.
+
+            Each Field Correction is labeled for a version of PLUTO. For programmatic changes, this is version in which the programmatic change was
+            implemented. For research and user reported changes, this is the version in which the BBL
+            change was added to PLUTO_input_research.csv.
+
+            For more information about the structure of the pluto corrections report,
+            see the [Pluto Changelog Readme](https://www1.nyc.gov/assets/planning/download/pdf/data-maps/open-data/pluto_change_file_readme.pdf?r=22v1).
+            """
+        )
+
+        if self.applied_corrections is None or self.not_applied_corrections is None:
+            st.info(
+                "There are no available corrections reports for this branch. This is likely due to a problem on the backend with the files on Digital Ocean."
+            )
+            return
+
+        version = st.sidebar.selectbox(
+            "Filter the field corrections by the PLUTO Version in which they were first introduced",
+            self.version_dropdown,
+        )
+
+        AppliedCorrectionsSection(self.applied_corrections, version)()
+        NotAppliedCorrectionsSection(self.not_applied_corrections, version)()
+
+        st.info(
+            """
+            See [here](https://www1.nyc.gov/site/planning/data-maps/open-data/dwn-pluto-mappluto.page) for a full accounting of the changes made for the latest version
+            in the PLUTO change file.
+            """
+        )
+
+
+class AppliedCorrectionsSection:
```

Comment:
The only thing I wonder here is if this is a good opportunity for class inheritance - ie: making a base CorrectionsSection class that includes the __init__ and all of the floating functions below, then these classes look just like:
```
  class AppliedCorrectionsSection(CorrectionsSection):
    def __call__(self):
       # stuff
```

Reply:
Before I get into the code here I wanted to share my thoughts on terminology. The terminology that seemed best to me while I was working on this yesterday is that each data product has a page, pages have reports (the corrections section is a report) and reports have sections. Our refactor here is bringing a report into a component file. Does this terminology seem right to you? Now is the time to think about it, future refactors into component files should follow whatever patterns we set here. 

In terms of this specific code, that inheritance would mean we don't have to pass `applied_corrections` or `not_applied_corrections` to the init of the section, we just reference `self.<appropriate corrections attr>` right? That would make the code neater, I'll make that change now